src/GraphQL.Tests/Types/Vector3ScalarTests.cs

```diff
@@ -0,0 +1,190 @@
+using System;
+using System.Collections.Generic;
+using GraphQL.Language.AST;
+using GraphQL.SystemTextJson;
+using GraphQL.Types;
+using Xunit;
+
+namespace GraphQL.Tests.Types
+{
+    // these tests relate to the code in custom-scalars.md
+    public class Vector3ScalarTests : QueryTestBase<Vector3ScalarTests.Vector3ScalarSchema>
+    {
+        [Fact]
+        public void test_parseliteral()
+        {
+            AssertQuerySuccess(@"{ input(arg: ""1,2,3"") }", @"{ ""input"": ""=1=2=3="" }");
+        }
+
+        [Fact]
+        public void test_parsevalue_string()
+        {
+            AssertQuerySuccess(@"query($arg: Vector3) { input(arg: $arg) }", @"{ ""input"": ""=1=2=3="" }", @"{ ""arg"": ""1,2,3"" }".ToInputs());
+        }
+
+        [Fact]
+        public void test_parsevalue_structured()
+        {
+            AssertQuerySuccess(@"query($arg: Vector3) { input(arg: $arg) }", @"{ ""input"": ""=1=2=3="" }", @"{ ""arg"": { ""x"": 1, ""y"": 2, ""z"": 3 } }".ToInputs());
+        }
+
+        [Fact]
+        public void test_default()
+        {
+            AssertQuerySuccess(@"{ input }", @"{ ""input"": ""=7=8=9="" }");
+        }
+
+        [Fact]
+        public void test_output()
+        {
+            AssertQuerySuccess(@"{ output }", @"{ ""output"": { ""x"": 4, ""y"": 5, ""z"": 6 } }");
```

Comment:
It works... thanks to the serializer. But I really doubt that the scalars were conceived to work like that.